### ISSUE TITLE
False-positive with PSR12.Operators.OperatorSpacing on PHP8

### ISSUE THREAD

**Issue by proton-ab** [squizlabs/PHP_CodeSniffer#3183]
**Code sample**
```php
<?php
declare(strict_types=1);

class Test
{
    public static function returnValue(?int $value = null): int|null
    {
        return $value;
    }
```

```
[x] Expected at least 1 space before "|"; 0 found (PSR12.Operators.OperatorSpacing.NoSpaceBefore)
```

**Expected behavior**
No issue should be detected here. This was previously fixed by https://github.com/squizlabs/PHP_CodeSniffer/issues/3151 but only for arguments.

**Versions (please complete the following information):**
 - OS: Debian 10
 - PHP: 8.0
 - PHPCS: 3.5.8
 - Standard: PSR12

**Comment by jrfnl** [comment]
> This was previously fixed by #3151 but only for arguments.

Untrue. This was fixed for both arguments as well as return types in one go, but as it [clearly states in the thread in #3151](https://github.com/squizlabs/PHP_CodeSniffer/issues/3151#issuecomment-719025793), the fix **will be in the PHPCS 3.6.0 release**.

Suggest: close.
